MovieApp/Views/DetailsScreen/MovieImageView.swift

```diff
@@ -94,6 +99,28 @@ class MoviewImageView: UIView{
         runTime.text = String(movieDetails.runtime / 60) + " h " + String(movieDetails.runtime % 60) + " m"
         runTime.textColor = .white
         runTime.font = UIFont.boldSystemFont(ofSize: 16)
+        
+        if(favorite){
+            favourite.setImage(UIImage(systemName: "star.fill"), for: .normal)
+        }else{
+            favourite.setImage(UIImage(systemName: "star"), for: .normal)
+        }
+            
+        favourite.layer.cornerRadius = 25
+        favourite.tintColor = .white
+        favourite.backgroundColor = UIColor(red: 0.459, green: 0.459, blue: 0.459, alpha: 0.6)
+        
+        favourite.addAction(.init{_ in
+            if(self.favorite){
+                self.moviesRepository.setMovieFavorite(id: Int32(self.movieDetails.id), favorite: false)
+                self.favourite.setImage(UIImage(systemName: "star"), for: .normal)
+                self.favorite = false
+            }else{
+                self.moviesRepository.setMovieFavorite(id: Int32(self.movieDetails.id), favorite: true)
+                self.favourite.setImage(UIImage(systemName: "star.fill"), for: .normal)
+                self.favorite = true
+            }
+        }, for: .touchUpInside)
```

Comment:
and with the previous comment you could

```swift
favourite.addAction(.init { [weak self] _ in
            guard let self = self else { return }

            self.moviesRepository.setMovieFavorite(id: Int32(self.movieDetails.id), favorite: !self.favorite)
            self.setFavoriteImage(isFavorite: !self.favorite)
            self.favorite = !self.favorite
        }, for: .touchUpInside)
```

Also notice that in you implementation you have introduced a memory leak, so keep in mind adding [weak self]
to closures that reference self